app/src/androidTest/java/de/test/antennapod/storage/DBWriterTest.java

```diff
@@ -121,6 +130,47 @@ public void testDeleteFeedMediaOfItemFileExists()
         assertNull(media.getFile_url());
     }
 
+    public void testDeleteFeedMediaOfItemRemoveFromQueue()
+            throws IOException, ExecutionException, InterruptedException, TimeoutException {
+        assertTrue(UserPreferences.shouldDeleteRemoveFromQueue());
+
+        File dest = new File(getInstrumentation().getTargetContext().getExternalFilesDir(TEST_FOLDER), "testFile");
+
+        assertTrue(dest.createNewFile());
+
+        Feed feed = new Feed("url", null, "title");
+        List<FeedItem> items = new ArrayList<>();
+        List<FeedItem> queue = new ArrayList<>();
+        feed.setItems(items);
+        FeedItem item = new FeedItem(0, "Item", "Item", "url", new Date(), FeedItem.UNPLAYED, feed);
+
+        FeedMedia media = new FeedMedia(0, item, 1, 1, 1, "mime_type", dest.getAbsolutePath(), "download_url", true, null, 0, 0);
+        item.setMedia(media);
+
+        items.add(item);
+        queue.add(item);
+
+        PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        adapter.setCompleteFeed(feed);
+        adapter.setQueue(queue);
+        adapter.close();
+        assertTrue(media.getId() != 0);
+        assertTrue(item.getId() != 0);
+        queue = DBReader.getQueue();
+        assertTrue(queue.size() != 0);
+
+        DBWriter.deleteFeedMediaOfItem(getInstrumentation().getTargetContext(), media.getId());
+        Thread.sleep(200);
```

Comment:
Maybe using Awaitility could be an option instead. The library is already included somewhere in AntennaPod.

    Awaitility.await().until(() -> dest.exists() == false);

It throws an exception if the condition is not met within 10 seconds, which then makes the test fail.